packages/discovery/src/discovery/permission-resolving/resolvePermissions.ts

```diff
@@ -23,7 +23,16 @@ import { assert, EthereumAddress } from '@l2beat/shared-pure'
 // what to do in a scenario when one members has a smaller delay than others
 // and such a case should be discussed.
 
-export type Permission = 'member' | 'act' | 'configure' | 'upgrade'
+export const RolePermissionEntries = [
```

Comment:
Please leverage TypeScript type system, so that I can't simply put an invalid value in `RolePermissionEntries`. Either decalre that values are from `RawPermissionConfiguration`, or - even better - define `RawPermissionConfiguration` as a union between RolePermissionEntries and e.g. NonRolePermissionEntries (act, configure, update).